test cases/rust/5 polyglot static fails with rust 1.55

**Describe the bug**
The test suite of meson (tested up to 0.59.2) fails in test cases/rust/5 polyglot static when using rust 1.55

```
[  246s] Found ninja-1.10.2 at /usr/bin/ninja
[  246s] [1/3] Compiling C object prog.p/prog.c.o
[  246s] [2/3] Compiling Rust source '../test cases/rust/5 polyglot static/stuff.rs'
[  246s] [3/3] Linking target prog
[  246s] FAILED: prog 
[  246s] cc  -o prog prog.p/prog.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libstuff.a -ldl -Wl,--end-group -pthread
[  246s] /usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: libstuff.a(std-d9afe58aec724292.std.fc180862-cgu.4.rcgu.o): in function `std::f32::<impl f32>::lerp':
[  246s] std.fc180862-cgu.4:(.text._ZN3std3f3221_$LT$impl$u20$f32$GT$4lerp17h54ca618b0a1b1fa1E+0x20): undefined reference to `fmaf'
[  246s] /usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: libstuff.a(std-d9afe58aec724292.std.fc180862-cgu.4.rcgu.o): in function `std::f64::<impl f64>::lerp':
[  246s] std.fc180862-cgu.4:(.text._ZN3std3f6421_$LT$impl$u20$f64$GT$4lerp17h0ff7a470044adf56E+0x22): undefined reference to `fma'
[  246s] collect2: error: ld returned 1 exit status
[  246s] ninja: build stopped: subcommand failed.
[  246s]  
[  246s] 
[  246s] ninja explain: deps for 'libstuff.a' are missing
[  246s] ninja explain: libstuff.a is dirty
[  246s] ninja explain: deps for 'prog.p/prog.c.o' are missing
[  246s] ninja explain: prog.p/prog.c.o is dirty
[  246s] ninja explain: libstuff.a is dirty
[  246s] ninja explain: prog is dirty
```

**To Reproduce**
Run the test suite with Rust 1.55

**Expected behavior**
A clear and concise description of what you expected to happen.

**system parameters**
* Tested in openSUSE Tumbleweed Staging, while attempting to upgrade to Rust 1.55
* Meson 0.59.2

--- a/test cases/rust/5 polyglot static/meson.build
+++ b/test cases/rust/5 polyglot static/meson.build
@@ -2,6 +2,7 @@ project('static rust and c polyglot executable', 'c', 'rust')
 
 deps = [
   meson.get_compiler('c').find_library('dl', required: false),
+  meson.get_compiler('c').find_library('m', required: false),
   dependency('threads'),
 ]
 

